frontend/src/views/Join.tsx

```diff
@@ -49,15 +58,21 @@ function JoinGamePage() {
             }}
             onKeyPress={(event) => {
               if (event.key === 'Enter' && validNickname) {
-                connect(SOCKET_ENDPOINT, nickname);
-                setPageState(2);
+                connect(SOCKET_ENDPOINT, nickname, setUsers).then(() => {
+                  setPageState(2);
+                }).catch((response) => {
+                  setError(response.message);
+                });
               }
             }}
           />
           <LargeInputButton
             onClick={() => {
-              connect(SOCKET_ENDPOINT, nickname);
-              setPageState(2);
+              connect(SOCKET_ENDPOINT, nickname, setUsers).then(() => {
+                setPageState(2);
+              }).catch((response) => {
+                setError(response.message);
+              });
```

Comment:
nit: Extract into a helper function since it seems like this is a duplicated code block

Reply:
Good idea -- done!